modules/oe_theme_inpage_navigation/tests/src/FunctionalJavascript/InPageNavigationBlockTest.php

```diff
@@ -0,0 +1,302 @@
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->container->get('entity_type.manager')
+      ->getStorage('node')
+      ->create([
+        'type' => 'oe_page',
+        'title' => 'Test Page node',
+        'oe_teaser' => 'Teaser text',
+        'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+        'uid' => 0,
+        'status' => 1,
+      ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    // Assert content part.
+    $this->assertSession()->elementExists('css', '#block-oe-theme-main-page-content[data-inpage-navigation-source-area] article');
+    // Assert absence of in-page navigation block.
+    $this->assertSession()->elementNotExists('css', '#block-inpage-navigation nav[data-ecl-inpage-navigation]');
+
+    InPageNavigationHelper::enableInPageNavigation($node);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    // Check that in-page navigation block still not visible.
+    $this->assertSession()->elementNotExists('css', 'nav[data-ecl-inpage-navigation]');
+
+    // Check that in-page navigation block visible when available
+    // heading in content.
+    $node->set('oe_related_links', [
+      [
+        'uri' => 'internal:/node',
+        'title' => 'Node listing',
+      ],
+      [
+        'uri' => 'https://example.com',
+        'title' => 'External link',
+      ],
+    ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    // Assert in-page navigation part.
+    $navigation = $this->assertSession()->elementExists('css', '#block-inpage-navigation nav[data-ecl-inpage-navigation]');
+    $inpage_nav_assert = new InPageNavigationAssert();
+    $inpage_nav_expected_values = [
+      'title' => 'Page contents',
+      'list' => [
+        ['label' => 'Related links', 'href' => '#related-links'],
+      ],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+
+    // Check that in-page navigation block visible when available
+    // heading in wysiwyg field.
+    $node->set('body', [
+      [
+        'value' => '<h2>Heading from body field</h2>',
+        'format' => 'full_html',
+      ],
+    ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    // Assert in-page navigation part.
+    $navigation = $this->assertSession()->elementExists('css', '#block-inpage-navigation nav[data-ecl-inpage-navigation]');
+    $inpage_nav_assert = new InPageNavigationAssert();
+    $inpage_nav_expected_values = [
+      'title' => 'Page contents',
+      'list' => [
+        ['label' => 'Heading from body field', 'href' => '#heading-from-body-field'],
+        ['label' => 'Related links', 'href' => '#related-links'],
+      ],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+
+    // Check that in-page navigation block count presence of
+    // h2.ecl-u-type-heading-2 elements inside content region.
```

Comment:
I don't think this should be tested like this. I'd add some custom markup in the test page template, and verify that everything is working. The fact that oe_theme uses this class for blocks should not matter in the test.

Reply:
I used block with heading not only for checking that it has an influence on in-page navigation but also insure that we have disappearing of in-page navigation block if we don't have anymore headings.